src/components/Upload.vue

```diff
@@ -1,66 +1,63 @@
 <template lang="pug">
 .upload-wrapper
   slot(name="toolbar")
-    v-toolbar(dense, dark, color="success")
-      v-toolbar-title Upload
+    v-toolbar(dense, dark, flat, color="primary")
+      v-icon cloud_upload
+      v-toolbar-title Upload to
+        v-avatar
+          v-icon {{ ResourceIcons[dest._modelType.toUpperCase()] }}
+        | {{ dest.name }}
       v-spacer
       v-btn(v-if="closeable", icon, flat, @click="$emit('close')")
         v-icon close
-  .px-0.py-2
-    slot(name="header")
-      .title.px-3
-        v-avatar.mr-2(color="blue-grey lighten-4")
-          v-icon {{ ResourceIcons[dest._modelType.toUpperCase()] }}
-        span {{ dest.name }}
 
   slot(name="dropzone")
     .dropzone-wrapper(
         v-if="!files.length", :class="dropzoneClass", @dragenter="dropzoneClass = 'animate'",
         @dragleave="dropzoneClass = null", @drop="dropzoneClass = null")
       .dropzone-message
         v-icon(size="50px") attach_file
-        .title.mt-3 {{ dropzoneMessage }}
+        .title {{ dropzoneMessage }}
```

Comment:
I think this was a good use of the margin class, I think it looks better to have a little space between the icon and line of text.

Reply:
Agreed. Was a bit overzealous here. I'll put it back.